### PR TITLE
[3.9.x][MNG-7586] Update Maven Resolver to 1.9.1

### PR DESCRIPTION
**Pull request by cstamas**
Update to Maven Resolver 1.9.1 and apply required changed changes in Maven.

Changes:
- hook in lifecycle

---

https://issues.apache.org/jira/browse/MNG-7586

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/internal/aether/ResolverLifecycle.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    private final Provider<RepositorySystem> repositorySystemProvider;
+
+    @Inject
+    public ResolverLifecycle( Provider<RepositorySystem> repositorySystemProvider )
+    {
+        this.repositorySystemProvider = requireNonNull( repositorySystemProvider );
+    }
+
+    @PreDestroy
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/internal/aether/ResolverLifecycle.java:49]: Now this is back?

Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/internal/aether/ResolverLifecycle.java:49]: Yes, we never removed it from Maven, but we did from Resolver

Comment (michael-o) [reply maven-core/src/main/java/org/apache/maven/internal/aether/ResolverLifecycle.java:49]: Ah, ok.

**Review by gnodet — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

**Comment by michael-o** [comment]
Shouldn't this be skipped for 1.9.1?

**Comment by cstamas** [comment]
recycled rather :smile:

**Comment by jira-importer** [comment]
Resolve #8877

**Pull request merged**